pkgs/tools/system/htop/default.nix

```
@@ -2,7 +2,8 @@
 , ncurses
 , IOKit
 , sensorsSupport ? stdenv.isLinux, lm_sensors
-, systemdSupport ? stdenv.isLinux, systemd
+, systemdSupport ? stdenv.isLinux && !stdenv.hostPlatform.isStatic
```

```suggestion
, systemdSupport ? stdenv.isLinux && lib.meta.availableOn stdenv.hostPlatform systemd
```
or
```suggestion
, systemdSupport ? stdenv.isLinux && stdenv.hostPlatform.libc == "glibc"
```
didn't tested but might need parenthesis.

Your check is clearly false because glibc can be static (see glibc.static)

---

It is not about libc, it is about static linking. Systemd builds on dynamic musl, but does not build statically at all, glibc or no glibc. See https://github.com/systemd/systemd/issues/20600#issuecomment-912338965

---

Ok my bad I thought about this https://lore.kernel.org/all/20190524172737.GA6813@localhost/T/